### PR TITLE
Add user defined labels to snapshot info

### PR DESCRIPTION
**Pull request by anmazzotti**
Part of #2172

### CODE REVIEW

**Comment by anmazzotti** [comment]
After install: `elemental install --debug --snapshot-labels myLabel=foo,myOtherLabel=bar`

```yaml
# Autogenerated file by elemental client, do not edit

date: "2024-09-09T08:43:38Z"
snapshotter:
    type: btrfs
    max-snaps: 4
    config: {}
efi:
    label: COS_GRUB
oem:
    label: COS_OEM
persistent:
    label: COS_PERSISTENT
recovery:
    label: COS_RECOVERY
    recovery:
        source: dir:///run/rootfsbase
        fs: squashfs
        labels:
            myLabel: foo
            myOtherLabel: bar
        date: "2024-09-09T08:43:38Z"
        fromAction: install
state:
    label: COS_STATE
    snapshots:
        1:
            source: dir:///run/rootfsbase
            active: true
            labels:
                myLabel: foo
                myOtherLabel: bar
            date: "2024-09-09T08:43:38Z"
            fromAction: install
```

After upgrade: `ELEMENTAL_UPGRADE_SNAPSHOT_LABELS="myLabel=bar,myOtherLabel=foo" elemental upgrade --debug --system oci://192.168.122.10:30000/elemental-os:dev-next`

```yaml
# Autogenerated file by elemental client, do not edit

date: "2024-09-09T08:45:57Z"
snapshotter:
    type: btrfs
    max-snaps: 4
    config: {}
efi:
    label: COS_GRUB
oem:
    label: COS_OEM
persistent:
    label: COS_PERSISTENT
recovery:
    label: COS_RECOVERY
    recovery:
        source: dir:///run/rootfsbase
        fs: squashfs
        labels:
            myLabel: foo
            myOtherLabel: bar
        date: "2024-09-09T08:43:38Z"
        fromAction: install
state:
    label: COS_STATE
    snapshots:
        1:
            source: dir:///run/rootfsbase
            labels:
                myLabel: foo
                myOtherLabel: bar
            date: "2024-09-09T08:43:38Z"
            fromAction: install
        2:
            source: oci://192.168.122.10:30000/elemental-os:dev-next
            digest: sha256:248b22b3dc24b730d1487df31c32fb527c40a95c74ea69c2d233d877a4e81d67
            active: true
            date: "2024-09-09T08:45:57Z"
            fromAction: upgrade
```

After upgrade+recovery: `ELEMENTAL_UPGRADE_SNAPSHOT_LABELS="myLabel=bar,myOtherLabel=foo" elemental upgrade --debug --system oci://192.168.122.10:30000/elemental-os:dev-next --recovery`

```yaml
# Autogenerated file by elemental client, do not edit

date: "2024-09-09T08:50:48Z"
snapshotter:
    type: btrfs
    max-snaps: 4
    config: {}
efi:
    label: COS_GRUB
oem:
    label: COS_OEM
persistent:
    label: COS_PERSISTENT
recovery:
    label: COS_RECOVERY
    recovery:
        source: dir:///run/rootfsbase
        fs: squashfs
        date: "2024-09-09T08:50:48Z"
        fromAction: upgrade
state:
    label: COS_STATE
    snapshots:
        1:
            source: dir:///run/rootfsbase
            labels:
                myLabel: foo
                myOtherLabel: bar
            date: "2024-09-09T08:43:38Z"
            fromAction: install
        2:
            source: oci://192.168.122.10:30000/elemental-os:dev-next
            digest: sha256:248b22b3dc24b730d1487df31c32fb527c40a95c74ea69c2d233d877a4e81d67
            date: "2024-09-09T08:45:57Z"
            fromAction: upgrade
        3:
            source: oci://192.168.122.10:30000/elemental-os:dev-next
            digest: sha256:248b22b3dc24b730d1487df31c32fb527c40a95c74ea69c2d233d877a4e81d67
            active: true
            date: "2024-09-09T08:50:48Z"
            fromAction: upgrade
```

After upgrade recovery only: `elemental upgrade-recovery --debug --snapshot-labels myLabel=foobar --recovery-system.uri oci://192.168.122.10:30000/elemental-os:dev-next`

```yaml
# Autogenerated file by elemental client, do not edit

date: "2024-09-09T08:54:49Z"
snapshotter:
    type: btrfs
    max-snaps: 4
    config: {}
efi:
    label: COS_GRUB
oem:
    label: COS_OEM
persistent:
    label: COS_PERSISTENT
recovery:
    label: COS_RECOVERY
    recovery:
        source: dir:///run/rootfsbase
        fs: squashfs
        labels:
            myLabel: foobar
        date: "2024-09-09T08:54:49Z"
        fromAction: upgrade-recovery
state:
    label: COS_STATE
    snapshots:
        1:
            source: dir:///run/rootfsbase
            labels:
                myLabel: foo
                myOtherLabel: bar
            date: "2024-09-09T08:43:38Z"
            fromAction: install
        2:
            source: oci://192.168.122.10:30000/elemental-os:dev-next
            digest: sha256:248b22b3dc24b730d1487df31c32fb527c40a95c74ea69c2d233d877a4e81d67
            date: "2024-09-09T08:45:57Z"
            fromAction: upgrade
        3:
            source: oci://192.168.122.10:30000/elemental-os:dev-next
            digest: sha256:248b22b3dc24b730d1487df31c32fb527c40a95c74ea69c2d233d877a4e81d67
            active: true
            date: "2024-09-09T08:50:48Z"
            fromAction: upgrade
```

After reset (from recovery partition): `elemental reset --debug --snapshot-labels myLabel=foobar,myOtherLabel=barfoo`

```yaml
# Autogenerated file by elemental client, do not edit

date: "2024-09-09T08:56:44Z"
snapshotter:
    type: btrfs
    max-snaps: 4
    config: {}
oem:
    label: COS_OEM
persistent:
    label: COS_PERSISTENT
recovery:
    label: COS_RECOVERY
    recovery:
        source: dir:///run/rootfsbase
        fs: squashfs
        labels:
            myLabel: foobar
        date: "2024-09-09T08:54:49Z"
        fromAction: upgrade-recovery
state:
    label: COS_STATE
    snapshots:
        1:
            source: dir:///run/rootfsbase
            active: true
            labels:
                myLabel: foobar
                myOtherLabel: barfoo
            date: "2024-09-09T08:56:44Z"
            fromAction: reset
```

**Comment by anmazzotti** [comment]
FYI the documentation regarding labels is missing, since the end-user is not going to be able to conveniently use them.
I'll add full documentation about labels with the `elemental state` feature.

**Comment by anmazzotti** [comment]
Noticed a bit late the environment variables were always empty, that needed a custom decoder, it's better now:

`ELEMENTAL_UPGRADE_SNAPSHOT_LABELS="myLabel=bar,myOtherLabel=foo" elemental upgrade --debug --system oci://192.168.122.10:30000/elemental-os:dev-next`

```yaml
# Autogenerated file by elemental client, do not edit

date: "2024-09-09T12:33:25Z"
snapshotter:
    type: btrfs
    max-snaps: 4
    config: {}
efi:
    label: COS_GRUB
oem:
    label: COS_OEM
persistent:
    label: COS_PERSISTENT
recovery:
    label: COS_RECOVERY
    recovery:
        source: dir:///run/rootfsbase
        fs: squashfs
        labels:
            myLabel: foo
            myOtherLabel: bar
        date: "2024-09-09T12:31:50Z"
        fromAction: install
state:
    label: COS_STATE
    snapshots:
        1:
            source: dir:///run/rootfsbase
            labels:
                myLabel: foo
                myOtherLabel: bar
            date: "2024-09-09T12:31:50Z"
            fromAction: install
        2:
            source: oci://192.168.122.10:30000/elemental-os:dev-next
            digest: sha256:11cb5c6f7b6b9e4daff67ec3ec7fa4a028c24445f2e4834a78e93c75d73eb5c3
            active: true
            labels:
                myLabel: bar
                myOtherLabel: foo
            date: "2024-09-09T12:33:25Z"
            fromAction: upgrade
```

File: cmd/flags.go [hunk 1]
```diff
@@ -74,6 +75,11 @@ func addRecoverySystemFlag(cmd *cobra.Command) {
 	cmd.Flags().String("recovery-system.uri", "", "Sets the recovery image source and its type (e.g. 'docker:registry.org/image:tag')")
 }
 
+// addSnapshotLabelsFlag adds labels to be applied to any newly created snapshot, applies to install, reset, upgrade, and upgrade-recovery actions.
+func addSnapshotLabelsFlag(cmd *cobra.Command) {
+	cmd.Flags().StringToString("snapshot-labels", map[string]string{}, "Add labels to the to the system (ex. --labels my-label=foo,my-other-label=bar)")
```
Comment (frelon) [inline cmd/flags.go:80]:
```suggestion
	cmd.Flags().StringToString("snapshot-labels", map[string]string{}, "Add labels to the to the system (ex. --snapshot-labels my-label=foo,my-other-label=bar)")
```

**Review by frelon — APPROVED** [review]
Just a small nit, looks good!

**Pull request merged**